### DISCUSSION TITLE
Trouble with Ubuntu 20.04.5 installs

### DISCUSSION THREAD

**Discussion by camberry** [automatic-ripping-machine/automatic-ripping-machine#652] [Q&A] [answered]
Not a bug report, but I thought I would use a bit of its format.

- [x] I have checked the wiki
- [X] I have searched for similar issues

**Describe the bug**
I am not a developer so forgive me if I am not filling this out perfectly.  I love using ARM, and it was working decently, but recently I have been getting errors when installing ARM to ubuntu for both the manual and the automatic script on both a physical machine and a VM.  Neither will work after installation. Multiple attempts for each type of installation. Results are the same.

Manual install:
After setup, the "history" and "database" (pasted image below) page throws an error.

Automatic Script:
After setup, the script does not work at all and does not appear to have finished.  Some folders (the /opt/arm/arm and /etc/arm I think).  Image also attached below of some errors after the script stops.

**To Reproduce**
1. Following the steps in the wiki for Ubuntu 20.04.5 Manual and auto
2. Leaving Postfix "Unconfigured."

**Environment**
[OS Distribution and version (run "cat /etc/lsb-release")]
Ubuntu 20.04.5 LTS
[ARM Release Version or if cloning from git branch/commit (run "git branch" and "git log -1" to get this)] 
v2.6.9 Stable

Automatic script install image: 
![Screenshot 2023-01-11 at 10 41 00 AM](https://user-images.githubusercontent.com/15239157/211882740-e5bd8e8b-f8d6-4192-8a4c-ff158f479a50.png)

Manual install Database page:
![Screenshot 2023-01-11 at 10 36 37 AM](https://user-images.githubusercontent.com/15239157/211882869-c02423e0-3d7c-4f23-b026-5fa22680acc0.png)

Web UI error log after manual install:
[WebUI.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/10394420/WebUI.log)

**Comment by shitwolfymakes** [comment]
Did you install this on a machine that did not already have an `arm` user?

**Reply by camberry** [reply]
Yeah, I tried it both ways. The instructions make an arm user in both the script and the manual. I have also tried making an arm user too and installing it from that user. Same results.

I installed it using the manual method about a year ago and it worked well and this time I can't get it to work. Curious that I have not seen anyone else with the same problem. Makes me think I might be doing something wrong.

**Reply by shitwolfymakes** [reply]
That may very well be. Maybe try spinning up a vm with a fresh install, or just reimage if that's an option?

**Comment by Kev1112005** [comment, marked as answer]
@camberry Have you verified that the Flask version that is installed on your machine is <3.X? I had similar issues (posted in #663) but resolved them by following the steps found in #596 

`pip install flask_sqlalchemy==2.5.1`

**Comment by dschulz72** [comment]
Not an expert on ARM, but given the above errors, I would try adding that directory `/home/arm/.local/bin` to your path. 

You can do this by opening the `.profile` file in /home/arm 
`sudo nano ~/.profile`

and adding the following line to the bottom of the file, on a new line:
`PATH='$PATH:/home/arm/.local/bin'`

Then source the file to load the new value:
`source ~/.profile`

Then try it again.

**Comment by shitwolfymakes** [comment]
Looks like I can fix this with changes to the installer script. I'll get to that today.
